src/content/developers/docs/data-structures/ssz/index.md

```diff
@@ -0,0 +1,153 @@
+
+```
+[37, 0, 0, 0, 55, 0, 0, 0, 16, 0, 0, 0, 22, 0, 0, 0, 1, 2, 3, 4]
+------------  -----------  -----------  -----------  ----------
+      |             |            |           |            |
+   number1       number2    offset for    number 3    value for
+                              vector                   vector
+
+```
+
+divided over lines for clarity:
+
+```
+[
+  37, 0, 0, 0,  # little-endian encoding of `number1`.
+  55, 0, 0, 0,  # little-endian encoding of `number2`.
+  16, 0, 0, 0,  # The "offset" that indicates where the value of `vector` starts (little-endian 16).
+  22, 0, 0, 0,  # little-endian encoding of `number3`.
+  1, 2, 3, 4,   # The actual values in `vector`.
+]
+```
+
+This is still a simplification - the integers and zeros in the schematics above would actually be stored bytelists, like this:
+
+```
+[
+  10100101000000000000000000000000  # little-endian encoding of `number1`
+  10110111000000000000000000000000  # little-endian encoding of `number2`.
+  10010000000000000000000000000000  # The "offset" that indicates where the value of `vector` starts (little-endian 16).
+  10010110000000000000000000000000  # little-endian encoding of `number3`.
+  10000001100000101000001110000100   # The actual value of the `bytes` field.
+]
+```
+
+So the actual values for variable-length types are stored in a heap at the end of the serialized object with their offsets stored in their correct positions in the ordered list of fields.
+
+There are also some special cases that require spoecific treatment, such as the `BitList` typoe that requires a length cap to be added during serialization and removed during deserialization. Full details are available in the [SSZ spec](https://github.com/ethereum/consensus-specs/blob/dev/ssz/simple-serialize.md).
+
+### Deserialization {#deserialization}
+
+To deserialize this object requires the <b>schema</b>. The schema defines the precise layout of the serialized data so that each specific element can be deserialized from a blob of bytes into some meaningful object with the elements having the right type, value, size and position. It is the schema that tells the deserializer which values are actual values and which ones are offsets. All field names disappear when an object is serialized, but reinstantiated on deserialization according to the schema.
+
+NB See [ssz.dev](https://www.ssz.dev/overview) for an interactive explainer on this.
+
+## Merkleization {#merkleization}
+
+This SSZ serialized object can then be merkleized - that is transformed into a Merkle-tree representation of the same data. First, the number of 32-byte chunks in the serialized object is determined. These are the "leaves" of the tree. The total number of leaves must be a power of 2 so that hashign together the leaves eventually produces a single hash-tree-root. If this is not naturally the case, additional leaves containing 32 bytes of zeros are added. Diagramatically:
```

Comment:
```suggestion
This SSZ serialized object can then be merkleized - that is transformed into a Merkle-tree representation of the same data. First, the number of 32-byte chunks in the serialized object is determined. These are the "leaves" of the tree. The total number of leaves must be a power of 2 so that hashing together the leaves eventually produces a single hash-tree-root. If this is not naturally the case, additional leaves containing 32 bytes of zeros are added. Diagrammatically:
```